File: src/webapp/frontend/types/draft.ts
export interface DraftPick {
  pick_id: number;
  draft_id: number;
  overall_pick: number;
  round_number: number;
  round_pick: number;
  team_id: number;
  player_id: number | null;
  original_team_id: number | null;
  trade_notes: string | null;
  college: string | null;
  height_in: number | null;
  weight_lbs: number | null;
  position: string | null;
  player_name?: string | null;
  team_abbrev?: string | null;
}

export interface Draft {
  draft_id: number;
  season_id: number;
  year: number;
  draft_date: string | null;
  round_count: number;
  pick_count: number;
  picks?: DraftPick[];
}

File: src/webapp/backend/app/schemas/draft.py
"""Draft Pydantic schemas for API request/response validation."""

from datetime import date
from pydantic import BaseModel, ConfigDict
from app.models.player import Position


class DraftPickBase(BaseModel):
    pick_id: int | None = None
    draft_id: int
    overall_pick: int
    round_number: int
    round_pick: int
    team_id: int
    player_id: int | None = None
    original_team_id: int | None = None
    trade_notes: str | None = None
    college: str | None = None
    height_in: int | None = None
    weight_lbs: int | None = None
    position: Position | None = None


class DraftPickSummary(DraftPickBase):
    model_config = ConfigDict(from_attributes=True)

    player_name: str | None = None
    team_abbrev: str | None = None


class DraftBase(BaseModel):
    draft_id: int | None = None
    season_id: int
    year: int
    draft_date: date | None = None
    round_count: int
    pick_count: int


class DraftSummary(DraftBase):
    model_config = ConfigDict(from_attributes=True)


class DraftDetail(DraftSummary):
    model_config = ConfigDict(from_attributes=True)

    picks: list[DraftPickSummary] = []
